### PR TITLE
Use flexBasis for changing splitter position

### LINKED ISSUE #47: Can't set the splitterposition after moving the splitter manually
_From @LennartKuell on December 11, 2018 8:48_

I tried to set the position in the code with .setSplitterPosition(90). This works fine, but it won't change the position, if I moved the splitter manually with my mouse before. So once I use the mouse to move the splitter, setSplitterPosition shows no effect anymore.
The reason seems to be, that moving the splitter with the mouse sets the "flex" attribute as style of the children, while setting it with setSplitterPosition sets the "height" attribute. I also tried to remove the flex attribute in the code, but this does not work, because the flex attribute isn't set in the children. I checked this with with splitLayout.getPrimaryComponent().getElement().getStyle().

I tested this with both Firefox and Chromium on LinuxMint and Chromium on Ubuntu.

_Copied from original issue: vaadin/vaadin-split-layout#126_

### PR DESCRIPTION
**Pull request by tulioag**
- Added connector with setSplitterPosition method that calculates and
sets flexBasis based on the percentage provided to setSplitterPosition
on server side.
- Replaced setting child element width with a call setSplitterPosition
method in connector.

Fixes #47

### CODE REVIEW

File: vaadin-split-layout-flow/src/main/resources/META-INF/resources/frontend/splitlayoutConnector.js [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations under
+ * the License.
+ */
+
+(function () {
+  const tryCatchWrapper = function (callback) {
+    return window.Vaadin.Flow.tryCatchWrapper(callback, 'Vaadin Split Layout', 'vaadin-split-layout-flow');
+  };
```
Comment (web-padawan) [inline vaadin-split-layout-flow/src/main/resources/META-INF/resources/frontend/splitlayoutConnector.js:20]: Note, if we want to backport this to `14.1` then we will have to remove this part there.

Comment (alvarezguille) [reply vaadin-split-layout-flow/src/main/resources/META-INF/resources/frontend/splitlayoutConnector.js:20]: Can this be backported at all?

**Comment by tulioag** [comment]
> Tested detaching and re-attaching the splitlayout in different roundtrips and the splitter position set with code is lost.
> This happens because the style added to the children is computed and applied in client-side and server side is not aware of it.

Fixed

**Comment by tulioag** [comment]
Should be closed if #77 is merged

**Comment by tulioag** [comment]
Closing in favor of #77

**Pull request closed without merging**